test/src/materials/LinearNonLinearIterationInterfaceMaterial.C

```diff
@@ -0,0 +1,40 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "LinearNonLinearIterationInterfaceMaterial.h"
+
+registerMooseObject("MooseTestApp", LinearNonLinearIterationInterfaceMaterial);
+
+defineLegacyParams(LinearNonLinearIterationInterfaceMaterial);
+
+InputParameters
+LinearNonLinearIterationInterfaceMaterial::validParams()
+{
+  InputParameters params = InterfaceMaterial::validParams();
+  params.addRequiredParam<MaterialPropertyName>("prop_name", "The name of the property");
+  params.addParam<Real>("prefactor", 1, "the prefactor to apply to the new material value");
+  params.addClassDescription(
+      "Interface material whose property is equal to (t_step+n_linear*n_nonlinear)*prefactor");
+  return params;
+}
+
+LinearNonLinearIterationInterfaceMaterial::LinearNonLinearIterationInterfaceMaterial(
+    const InputParameters & parameters)
+  : InterfaceMaterial(parameters),
+    _mat_prop(declareProperty<Real>(getParam<MaterialPropertyName>("prop_name"))),
+    _prefactor(getParam<Real>("prefactor"))
+{
+}
+
+void
+LinearNonLinearIterationInterfaceMaterial::computeQpProperties()
+{
+  _mat_prop[_qp] =
+      (_t_step + _fe_problem.nNonlinearIterations() * _fe_problem.nLinearIterations()) * _prefactor;
```

Comment:
This is a bit scary. Are you doing `Exodiff` or similar on a field related to this? If we try to update PETSc and somehow a solve takes a different number of iterations, then this would have to be regolded and/or we'd have to slap a specific petsc_version number on this test

Reply:
@lindsayad I see your point, however I thought about it. This material is. used in a test that only errors out if and when the material property is not equal to the theoretical material property compute inside the test user object. The test uses a `RunApp` spec and is completely independent from the obtained solution. As long as the test converges and no error are generated this is fine.

Reply:
Excellent!